internal/loader/configuration_setting_loader.go

```diff
@@ -245,7 +245,7 @@ func (csl *ConfigurationSettingLoader) CreateKeyValueSettings(ctx context.Contex
 			}
 
 			currentUrl := *setting.Value
-			secretUriSegment, err := parse(currentUrl)
+			secretMetadata, err := parse(currentUrl)
```

Comment:
This line and line200 are all using the secretMatadata word, that indeed causes confusion. I would suggest `SecretMatadatas`(regardless the plural and singular same form) or `SecretsMetadata` in `TargetSecretReference`